shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/resources/BaseResourceStatus.java

```diff
@@ -0,0 +1,31 @@
+package com.redhat.service.smartevents.shard.operator.v2.resources;
+
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.redhat.service.smartevents.infra.core.exceptions.BridgeError;
+import com.redhat.service.smartevents.infra.v2.api.models.DefaultConditions;
+import com.redhat.service.smartevents.shard.operator.core.resources.Condition;
+import com.redhat.service.smartevents.shard.operator.core.resources.ConditionStatus;
+import com.redhat.service.smartevents.shard.operator.core.resources.CustomResourceStatus;
+
+abstract class BaseResourceStatus extends CustomResourceStatus {
+
+    public BaseResourceStatus(Set<Condition> initialConditions) {
+        super(initialConditions);
+    }
+
+    @JsonIgnore
+    public final boolean isReady() {
+        return getConditions().stream().allMatch(c -> ConditionStatus.True.equals(c.getStatus()));
+    }
+
+    @JsonIgnore
+    public final void setStatusFromBridgeError(BridgeError bridgeError) {
+        markConditionFailed(DefaultConditions.CP_DATA_PLANE_READY_NAME,
```

Comment:
@lucamolteni This `Condition` name may need changing depending on what you do for "timeout".

Reply:
@manstis Please correct me if I'm wrong, but this is related to `Exception` handling and not time-out. See https://javaoperatorsdk.io/docs/features#automatic-retries-on-error and https://javaoperatorsdk.io/docs/features#setting-error-status-after-last-retry-attempt . We don't throw an `Exception` when we time out, we just mark a condition as `Failed` (or will do). Equally the `Exception` retry limit could be reached before the timeout limit.

In any case, we would need to do something like:

1. Provide a mapping between a specific `Condition` and `BridgeError` somehow; or 
2. Have the Controller decide in the implementation of `updateErrorStatus` which `Condition` should be marked as `Failed`

I don't think we have a ticket for mapping `Exception` to `Condition`, so I'd suggest we remove this from `v2` , create the ticket for mapping `Exception` to `Condition` and leave the `updateErrorStatus` method empty within the `ManagedProcessorController`.

Reply:
@robpblake Correct.. but the `v2` Controllers have no exception handling really.

[One](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedProcessorController.java#L146-L154) has some legacy code from how we did it in `v1` and the [other](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedBridgeController.java#L161-L164) nothing.

How exceptions are propagated from the `DataPlane` to the `ControlPlane` in `v2` remains erm, undetermined.

The existing code set `ConditionTypeConstants.READY` to `False`. Which is completely wrong in `v2`.

So I made the change in this PR and [suggested](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1492/files#r1072141411) to @lucamolteni it would need review.

Reply:
I can however remove the legacy use in `ManagedProcessorController`, as you propose, and create a ticket for the error mapping, also as you propose. :+1:

Reply:
New commit added.

Reply:
https://issues.redhat.com/browse/MGDOBR-1412